doc/primer.rst

```diff
@@ -165,6 +168,410 @@ iterates on each member of the stack.
+        pendingWires: []
+        pendingEdges: []
+        tags: {}
+
+This is simply an empty :class:`~cadquery.Workplane`. Being the first :class:`~cadquery.Workplane`
+in the chain, it does not have a parent. The :attr:`~cadquery.Workplane.plane` attribute contains a
+:class:`~cadquery.Plane` object that describes the XY plane.
+
+Now we create a simple box. To keep things short, the ``print(part)`` line will not be shown for the
+rest of these code blocks::
+
+    part = part.box(1, 1, 1)
+
+Which produces the output:
+
+.. code-block:: none
+
+    Workplane object at 0x7f5c368eaa90:
+      parent: Workplane object at 0x7f5c9b822760
+      plane: Plane object at 0x7f5c368f3850:
+        origin: (0.0, 0.0, 0.0)
+        z direction: (0.0, 0.0, 1.0)
+      objects: [<cadquery.occ_impl.shapes.Solid object at 0x7f5c9b94bbe0>]
+      modelling context: CQContext object at 0x7f5c9b822730:
+        pendingWires: []
+        pendingEdges: []
+        tags: {}
+
+The first thing to note is that this is a different :class:`~cadquery.Workplane` object to the
+previous one, and in the :attr:`~cadquery.Workplane.parent` attribute of this
+:class:`~cadquery.Workplane` is our previous :class:`~cadquery.Workplane`. Secondly, the modelling
+context object is the same as the one in the previous :class:`~cadquery.Workplane`, and this one
+modelling context will be shared between every :class:`Workplane` object in this chain. Thirdly, in
+our objects list is a single :class:`~cadquery.Solid` object, which is the box we just created.
+
+Often when creating models you will find yourself wanting to refer back to a specific
+:class:`~cadquery.Workplane` object, perhaps because it is easier to select the feature you want in this
+earlier state, or because you want to reuse a plane. Tags offer a way to refer back to a previous
+:class:`~cadquery.Workplane`. We can tag the :class:`~cadquery.Workplane` that contains this basic box now::
+
+    part = part.tag("base")
+
+The string representation of ``part`` is now:
+
+.. code-block:: none
+
+    Workplane object at 0x7f5c368eaa90:
+      parent: Workplane object at 0x7f5c9b822760
+      plane: Plane object at 0x7f5c368f3850:
+        origin: (0.0, 0.0, 0.0)
+        z direction: (0.0, 0.0, 1.0)
+      objects: [<cadquery.occ_impl.shapes.Solid object at 0x7f5c9b94bbe0>]
+      modelling context: CQContext object at 0x7f5c9b822730:
+        pendingWires: []
+        pendingEdges: []
+        tags: {'base': <cadquery.cq.Workplane object at 0x7f5c368eaa90>}
+
+The :attr:`~cadquery.cq.CQContext.tags` attribute of the modelling context is simply a dict
+associating the string name given by the :meth:`~cadquery.Workplane.tag` method to the
+:class:`~cadquery.Workplane`. Methods such as :meth:`~cadquery.Workplane.workplaneFromTagged` and
+selection methods like :meth:`~cadquery.Workplane.edges` can operate on a tagged
+:class:`~cadquery.Workplane`. Note that the :meth:`~cadquery.Workplane.tag` method is unusual for a
```

Comment:
I don't really understand the "Note",  Maybe adding an example here would make it more clear

Reply:
Thanks @marcus7070 from the questions that have been asked in the discord I feel that a lot of them would be resolved by the reading of this part of the docs !

Reply:
3c43519503cb42a74e31acd482d766ef0fa1bb61 is aimed at making this more clear, does it help?